codebasin/__init__.py

```diff
@@ -123,3 +127,112 @@ def from_json(cls, instance: dict):
             command=command,
             output=output,
         )
+
+
+class CodeBase:
+    """
+    A representation of all source files in the code base.
+
+    Attributes
+    ----------
+    directories: list[str | os.PathLike[str]]
+        The set of source directories that make up the code base.
+
+    exclude_patterns: list[str]
+        A set of patterns describing source files excluded from the code base.
+    """
+
+    def __init__(
+        self,
+        *directories: str | os.PathLike[str],
+        exclude_patterns: list[str] = [],
```

Comment:
Should this be an `Iterable[str]`? Or do we specifically use list functionality?

Reply:
I don't think we're using any functionality of `list` right now, so I've swapped this out for `Iterable` in 55a62ce9fcf918464651bec9a64f0400fb888a5f.